bindings/python/pymongocrypt/mongocrypt.py

```diff
@@ -490,11 +502,11 @@ def __init__(self, ctx, kms_provider, opts, callback):
         """
         super(DataKeyContext, self).__init__(ctx)
         try:
-            if kms_provider not in ['aws', 'gcp', 'azure', 'local']:
+            if kms_provider not in ['aws', 'gcp', 'azure', 'kmip', 'local']:
```

Comment:
This is the actual change that we need to allow "kmip" to be passed. The rest of the changes in this PR are docs and misc cleanup.